File: Sorting And Searching/Find Majority Element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int elem;
        int count=0;
        int n=nums.size();
        for(int i=0;i<n;i++){
            if(count==0){
                elem=nums[i];
            }
            if(nums[i]==elem){
                count++;
            }
            else{
                count--;
            }
            
        }
        return elem;
    }
};

File: searchingAndSorting/find_majority_element.java
class Solution
{
    static int majorityElement(int a[], int size)
    {
        // your code here
        int max=a[0];
        for (int i=0;i<size;i++){
            if(max<a[i]){
                max=a[i];
            }
        }
        
        int []freq=new int[max+1];
        
        for (int i=0;i<size;i++){
            freq[a[i]]++;
        }
        
        int i;
        for (i=0;i<max+1;i++){
            if(freq[i]>size/2 & freq[i]>0 ){
                return i;
            }
        }
       // System.out.print(size/2);
        
        return -1;
    }
}
